text/0009-golang.md

```diff
@@ -0,0 +1,70 @@
+ 
+The simplicity of Go stems from the fact that, unlike Java, it is not a multi-purpose tool. Go's main uses cases are web services and CLIs (command-line interfaces) which makes it a perfect fit for writing FOLIO backend modules. Being simpler than Java, the language is also easier to pick up by full-stack developers coming from scripting and web languages background like Node.js or Python. This can be a big advantage for small teams of few developers as the entire team can, effectively, be kept full-stack. With large Java codebases it’s not practical and developers in the team usually specialize in front-end or back-end programming. 
+
+On the operational front, Go's characteristics make it a better fit than Java for building systems comprised of many small microservices. Go is compiled ahead-of-time (AOT) into a static binary which is a perfect fit for Docker containers. AOT compilation means that the resulting application has a visibly smaller memory footprint and faster startup times as compared to Java/vert.x, and orders of magnitude better as compared to Java/Spring based modules. Java provides AOT compilation capability through the GraalVM project, but porting existing Java projects to GraalVM involves substantial complexity and is not possible for certain frameworks like older Spring (including versions used by FOLIO project) and Grails due to the dynamic language features they require. 
+
+From a sysops perspective, Go containers require less operational tuning compared to the JVM containers as they do not contain a stand-alone virtual machine that requires parametrization.
+
+The Go language is widely used across the industry and open-source communities, especially in the areas related to web and microservices. Popular examples of large open-source projects created with Go include the [Kubernetes](https://github.com/kubernetes/kubernetes) container orchestration platform and the [Docker](https://github.com/docker) container runtime.
+
+The Go project maintains a [list](https://go.dev/solutions/) of case-studies from commercial companies relying on Go in their products and general use-cases for the language.
+
+According to [GitHub statistics](https://madnight.github.io/githut/#/pull_requests/2023/3), Go is the third fastest growing language (behind Python and Java) as measured by the number of new Pull Requests.
+
+## Risks and Drawbacks
```

Comment:
Another consideration is tooling.  Folio uses several tools such as Snyk and Sonarqube.  Do these support Go?  If not, the community will need to find, adopt, maintain and incorporate alternatives into their existing processes.

Reply:
The timing section below suggests a pilot project.  Assuming that is meant to be a PoC, it should be done in a way to answer these questions on tooling as well.

Reply:
@craigmcnally @maccabeelevine Agree, this should be answered as part of the pilot project.